qutebrowser/completion/completionwidget.py

```diff
@@ -255,3 +257,26 @@ def showEvent(self, e):
         if scrollbar is not None:
             scrollbar.setValue(scrollbar.minimum())
         super().showEvent(e)
+
+    def hide(self):
+        super().hide()
+
+        if config.get('completion', 'auto-open') == True:
+            return
+
+        cmd = objreg.get('status-command', scope='window', window=self._win_id)
+        try:
+            cmd.cursorPositionChanged.disconnect(cmd.update_completion)
+            cmd.textChanged.disconnect(cmd.update_completion)
+        # Don't fail if not connected
+        except TypeError:
+            pass
+
+    def _open_completion_if_needed(self):
+        if config.get('completion', 'auto-open') == False:
+            update_completion = pyqtSignal()
```

Comment:
Wait, it's assigning self.update_completion... I've been programming Ruby for over a year, so excuse the confusion O.o